### PR TITLE
feat: don't manage Konnect internal-only plugins

### PR DESCRIPTION
**Pull request by GGabriele**
Plugins created by Konnect for internal-use only
are tagged with 'konnect-managed-plugin'. decK should not allow users to
interact with any plugins with this tag.

This skips plugins with internal-only konnect tags.

### CODE REVIEW

File: dump/dump_konnect.go [hunk 1]
```diff
@@ -168,3 +169,21 @@ func kongServiceIDs(cpID string,
...
 }
+
+// excludeKonnectManagedPlugins filter out konnect-managed plugins
+func excludeKonnectManagedPlugins(plugins []*kong.Plugin) []*kong.Plugin {
+	var filtered []*kong.Plugin
+	for _, p := range plugins {
+		var konnectManaged bool
+		for _, t := range p.Tags {
+			if *t == konnect.KonnectManagedPluginTag {
+				konnectManaged = true
```
Comment (hbagdi) [inline dump/dump_konnect.go:180]:
```suggestion
				konnectManaged = true
				break
```

File: dump/dump_konnect.go [hunk 2]
```diff
@@ -168,3 +169,21 @@ func kongServiceIDs(cpID string,
 	}
 	return res
 }
+
+// excludeKonnectManagedPlugins filter out konnect-managed plugins
```
Comment (hbagdi) [inline dump/dump_konnect.go:173]:
```suggestion
// excludeKonnectManagedPlugins filters out konnect-managed plugins.
```

File: dump/dump_konnect_test.go [hunk 3]
```diff
@@ -280,3 +280,63 @@ func Test_filterNonKongPackages(t *testing.T) {
...
 }
+
+func Test_excludeKonnectManagedPlugins(t *testing.T) {
+	tests := []struct {
+		name    string
+		plugins []*kong.Plugin
+		want    []*kong.Plugin
+	}{
+		{
+			name: "eclude konnect tags",
```
Comment (hbagdi) [inline dump/dump_konnect_test.go:291]:
```suggestion
			name: "exclude konnect tags",
```

We should enable a spell check linter to automate this.

File: dump/dump_konnect_test.go [hunk 4]
```diff
@@ -280,3 +280,63 @@ func Test_filterNonKongPackages(t *testing.T) {
...
+				{
+					Name: kong.String("rate-limiting"),
+					Tags: []*string{kong.String("tag1")},
+				},
+				{
+					Name: kong.String("basic-auth"),
+					Tags: []*string{},
+				},
+			},
+		},
```
Comment (hbagdi) [inline dump/dump_konnect_test.go:332]: Add test cases:
1. Empty array input
2. Non-empty input but empty result (all plugins managed)

**Review by hbagdi — COMMENTED** [review]
Needs a bit of clean up but LGTM otherwise.

The nested for loop would look better if the logic to detect the tag was in a separate function.

File: dump/dump_konnect_test.go (hunk 4 above)
Comment (GGabriele) [reply dump/dump_konnect_test.go:332]: Done.

**Review by hbagdi — APPROVED** [review]

**Pull request merged**